### PR TITLE
feat: pre eip155 transactions

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 0.4 day

Resolves: #1098 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Support white listed pre eip 155 transactions.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: docker-compose.prod.yaml [hunk 1]
```diff
@@ -81,6 +81,7 @@ services:
       - ACCOUNT_CONTRACT_CLASS_HASH=0x3c1081f98ecdd1d320368d2a136191a3fbf02a3e419463d77c3c5aa6dbc99a0
       - MAX_FELTS_IN_CALLDATA=30000
       - RETRY_TX_INTERVAL=2
+      - WHITE_LISTED_EIP_155_ADDRESS=
```
Comment (tcoratger) [inline docker-compose.prod.yaml:84]: I think we should put this also in the env example file no?

File: src/eth_provider/error.rs [hunk 2]
```diff
@@ -219,6 +219,9 @@ pub enum TransactionError {
     /// Thrown when the chain id is invalid.
     #[error("invalid chain id")]
     InvalidChainId,
+    /// Thrown when the transaction type is invalid.
+    #[error("invalid transaction type")]
+    InvalidTransactionType,
```
Comment (tcoratger) [inline src/eth_provider/error.rs:224]: As a followup good first issue, maybe we can improve some of these errors. For example:
- `InvalidTransactionType` could come with an error message indicating precisely the transaction type in question.
- I think it's the same for `InvalidChainId` where we could put an error message saying: "expected chain ID: ...., obtained chain ID: ...`

in order to give more information to these errors. If you agree I can open an issue about this.

File: tests/tests/eth_provider.rs [hunk 3]
```diff
@@ -630,6 +632,59 @@ async fn test_send_raw_transaction(#[future] katana: Katana, _setup: ()) {
...
+    let bytes = tx_hash.0;
+    let starknet_tx_hash = FieldElement::from_bytes_be(&bytes).unwrap();
+
+    watch_tx(eth_provider.starknet_provider(), starknet_tx_hash, std::time::Duration::from_millis(300), 60)
+        .await
+        .expect("Tx polling failed");
+
+    // Then
+    let count = eth_provider.storage_at(counter_address, JsonStorageKey::from(U256::from(0)), None).await.unwrap();
+    assert_eq!(B256::left_padding_from(&[0x1]), count);
```
Comment (tcoratger) [inline tests/tests/eth_provider.rs:685]: Small nit, I think the expected value should be on the right hand side no?

File: src/eth_provider/error.rs (hunk 2 above)
Comment (greged93) [reply src/eth_provider/error.rs:224]: Agreed!

Comment (tcoratger) [reply src/eth_provider/error.rs:224]: Done https://github.com/kkrt-labs/kakarot-rpc/issues/1112

File: docker-compose.yaml [hunk 4]
```diff
@@ -64,10 +64,9 @@ services:
         condition: service_completed_successfully
 
   kakarot-rpc:
-    image: ghcr.io/kkrt-labs/kakarot-rpc/node:latest
+    image: greged93/rpc:latest
```
Comment (tcoratger) [inline docker-compose.yaml:67]:
```suggestion
    image: ghcr.io/kkrt-labs/kakarot-rpc/node:latest
```

File: docker-compose.yaml [hunk 5]
```diff
@@ -64,10 +64,9 @@ services:
...
     # Always pull the latest image, until we use release tags
     # TODO: use release tags
-    pull_policy: always
```
Comment (tcoratger) [inline docker-compose.yaml:70]: Why should we remove this?

**Review by tcoratger — APPROVED** [review]

**Pull request merged**